adapter.js

```diff
@@ -1,48 +1,96 @@
 // deno-lint-ignore-file no-unused-vars
+import { Queue, R } from "./deps.js";
 
-export default function ({ db }) {
+const { pluck, omit } = R;
+const queue = new Queue();
 
-  async function index() {
-    return db.find({})
+export default function ({ db }) {
+  function index() {
+    return db.find({ type: "queue" })
+      .then(pluck("name"));
   }
 
-  async function create({ name, target, secret }) {
+  function create({ name, target, secret }) {
     return db
-      .insert({ _id: name, name, target, secret })
-      .then(doc => ({ ok: true, _id: doc._id }))
+      .insert({ type: "queue", _id: name, name, target, secret })
+      .then((doc) => ({ ok: true, _id: doc._id }));
   }
 
-  async function doDelete(name) {
+  function doDelete(name) {
     return db
       .removeOne({ _id: name })
-      .then(_doc => ({ ok: true }))
+      .then((_doc) => ({ ok: true }));
   }
 
   async function post({ name, job }) {
     // TODO
-    return Promise.resolve({ ok: true })
+    job = { queue: name, type: "job", ...job, status: "READY" };
+    // get queue data
+    const q = await db.findOne({ _id: name });
+    // store job doc to db
+    job = await db.insert(job);
+    // create job post function
+    // push job post function to queue
+    queue.push(async () =>
+      await postJob(q, job)
+        .then((result) =>
+          db.updateOne({ _id: job._id }, { $set: { status: result.status } })
+        )
+        .catch((e) =>
+          db.updateOne({ _id: job._id }, { $set: { status: "ERROR" } })
+        )
+      //.then(console.log.bind(console))
+    );
+    // return success response
+    return Promise.resolve({ ok: true, _id: job._id });
```

Comment:
The `Queue` port accepts an `id`, should also return `id` and not `_id`? The `Data` port returns `id` not `_id`. So  consistently will lend itself to less of an api surface for consumers to remember.